### PR TITLE
AST Transform for Nested Invocation with ::

### PR DESCRIPTION
**Pull request by rtablada**

### CODE REVIEW

**Comment by rtablada** [comment]
@rwjblue yeah I didn't see that I only had run default try scenario and didn't catch. I wondered if the AST was noop in 3.4+.

Updating to the two AST approach.

File: index.js [hunk 1]
```diff
@@ -46,6 +55,16 @@ module.exports = {
     };
   },
 
+  _buildNestedPlugin() {
+    return {
+      name: 'component-attributes',
```
Comment (rwjblue) [inline index.js]:
```suggestion
      name: 'nested-component-invocation-support',
```

File: lib/ast-nested-transform.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    function replaceNestedComponents(string) {
+      return string.replace('::', '/');
+    }
+
+    function getBlockParamName(node) {
+      let unnestedName = node.tag.replace('::', '');
+      let possibleName = unnestedName;
+
+      const nestedNames = node.children
```
Comment (rwjblue) [inline lib/ast-nested-transform.js]: I don't think this guard is quite correct, there are a few cases not accounted for:

* the given name _could_ be used by any number of nested children below us (this only checks direct children)
* this is only checking for invocations using the possible name, but not for things like path expressions, yielded block params, etc
* the name we pick could be clobbering a name that is in use in the outer scope (e.g. above us)
* the logic doesn't currently support `<Foo::Bar::Baz />` (`string.replace('::', '/')` only replaces the first occurrence of `::`)

Also, the logic to add the `let` will actually add many more `let` blocks than we might want. Consider:

```hbs
<Foo::Bar />
<Foo::Bar />
<Foo::Bar />
<Foo::Bar />
<Foo::Bar />
<Foo::Bar />
```

With the current transform, ^ template would end up with many "extra" `let` blocks.

**Comment by rwjblue** [comment]
@rtablada - Ready for review, lemme know what you think...

Also, you can poke at the transform and see it "live" here:

https://astexplorer.net/#/gist/0590eb883edfcd163b183514df4cc717

File: lib/ast-nested-transform.js (hunk 2 above)
Comment (rwjblue) [reply lib/ast-nested-transform.js]: I believe the most recent round of updates that I just pushed address all of these concerns...

**Comment by rtablada** [comment]
@rwjblue looks good! I used the ast transformer to debug live to build this!

**Pull request merged**